src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/core/stats/collectors/samplers/SampleCollector.java

```diff
@@ -0,0 +1,85 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.collectors.samplers;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.collectors.Collector;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.eval.Statistics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.eval.impl.Sample;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.eval.impl.vals.Value;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.format.Formatter;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.measurements.MeasurementSet;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.measurements.sampled.SampledMeasurements;
+import com.google.common.collect.ImmutableMap;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class SampleCollector implements Collector {
+  private static final Logger LOG = LogManager.getLogger(SampleCollector.class);
+  /** These are the measurement types this collector keeps track of. */
+  private final SampledMeasurements[] sampleSet;
+
+  /** When was the first updateStat was called since the last reset. */
+  private AtomicLong startTimeMillis;
+
+  /** This map keeps track of the samples corresponding to each measurement type. */
+  private ImmutableMap<SampledMeasurements, Sample> measurementSampleValueMap;
+
+  public SampleCollector(SampledMeasurements[] sampleSet) {
```

Comment:
We can merge this with SampleAggregator.

Reply:
done